### ISSUE TITLE
Directory links from pollen files

### ISSUE THREAD

**Issue by jnboehm** [mbutterick/pollen-users#72]
Most blog engines form links from blog posts routing to a directory (i. e. `example.com/post/my-title/`), whereas pollen has a titled html file as the result (so `example.com/post/my-title.html`).  Has there been effforts or thoughts on creating canonical links in a similar vein as the first example with pollen?

I personally prefer that option as it keeps the result language agnostic, despite serving the same html file content (some [supporting blog posts](https://www.hanselman.com/blog/urls-are-ui)).

The approach taken for this would essentially boil down to creating the directory in question and generating the html file there (like `example.com/post/my-title/index.html`).  This would change the meaning of a `.poly.pm` or `.html.pm` file so that would also be an argument against this behavior.

**Comment by mbutterick** [comment]
> Has there been effforts or thoughts on creating canonical links in a similar vein as the first example with pollen?

No, because this is a form of magic performed by a web server, and Pollen has a strong policy against magical thinking. In any case, it seems unnecessary: why not write a `link` tag function that omits the trailing `index.html` for URLs inside the `/post/` directory.

**Comment by jnboehm** [comment]
Huh, interesting, I always assumed that it would be a default from the browser, never questioned it too much.

Thank you for clarifying the rationale behind it!

**Comment by jnboehm** [comment]
A follow-up question:  do you (does anyone) have a good way to work around this in the light of the index.ptree file?  Would it be safe to rename the `.html` extension to a trailing slash to indicate the directory?

**Comment by mbutterick** [comment]
I wouldn’t dissuade anyone from experimenting. Consider, however, that the pagetree also cooperates with `raco pollen render ···` to render certain files on disk. For that reason, it usually makes sense for the pagetree to refer to local (output) file paths. If you don’t care about this aspect of things, hack away.
